### PR TITLE
Basic cursor null ref and bitwise fixes

### PR DESCRIPTION
**Pull request by HodgsonSDAS**
Fixes https://github.com/Microsoft/HoloToolkit-Unity/issues/246 Null Ref on Awake when looking for `GazeManager.Instance`.
Fixes https://github.com/Microsoft/HoloToolkit-Unity/issues/197 Layer raycast mask to use bitwise op for calculation.

### CODE REVIEW

File: Assets/HoloToolkit/Input/Scripts/BasicCursor.cs [hunk 1]
```diff
@@ -28,15 +28,11 @@ public struct RaycastResult
 
         private bool hasLoggedGazeManagerError;
 
+        private GazeManager gazeManagerReference;
```
Comment (NeerajW) [inline Assets/HoloToolkit/Input/Scripts/BasicCursor.cs]: Lets call this gazeManager or gazeManagerInstance to be consistent with other classes.

File: Assets/HoloToolkit/Input/Scripts/BasicCursor.cs [hunk 2]
```diff
@@ -48,7 +44,25 @@ protected virtual void Awake()
...
+            cursorDefaultRotation = gameObject.transform.rotation;
+        }
+
+        private bool GetGazeManagerReference()
+        {
+            gazeManagerReference = GazeManager.Instance;
+
+            if (gazeManagerReference == null)
+            {
+                return false;
```
Comment (NeerajW) [inline Assets/HoloToolkit/Input/Scripts/BasicCursor.cs]: Debug.LogError(missing GazeManager in the scene)

Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/BasicCursor.cs]: I didn't want to spam this in case the reference is just null for a few frames.

Comment (NeerajW) [inline Assets/HoloToolkit/Input/Scripts/BasicCursor.cs]: I see your motivation and concern. But this is a very common mistake.

Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/BasicCursor.cs]: I could count count 5 or 10 loops before throwing the error, what do you think?

Comment (keveleigh) [inline Assets/HoloToolkit/Input/Scripts/BasicCursor.cs]: I'd prefer changing Awake to Start. If you're worried about inherited cases, we could also move only the `if ((GazeManager.Instance.RaycastLayerMask...` etc part into Start and leave the rest of Awake alone.

If we choose to go the route where this code is moved into LateUpdate, I don't think we have to wait any frames before throwing the error. Have you run into a case where the reference is null for a few frames? As soon as GazeManager's GameObject is instantiated (which would happen before this script's Start and way before LateUpdate), FindObjectOfType will return non-null.

Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/BasicCursor.cs]: Even better idea!

Comment (keveleigh) [inline Assets/HoloToolkit/Input/Scripts/BasicCursor.cs]: Haha, I do get lucky with an idea occasionally! 😜 Thanks for making this change!

**Pull request merged**